loader/unknown_ext_chain_gas_aarch64.S

```diff
@@ -0,0 +1,833 @@
+//
+// Copyright (c) 2021 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Eric Sullivan <esullivan@nvidia.com>
+//
+
+// This code is used to pass on device (including physical device) extensions through the call chain. It must do this without
+// creating a stack frame, because the actual parameters of the call are not known. Since the first parameter is known to be a
+// VkPhysicalDevice or a dispatchable object it can unwrap the object, possibly overwriting the wrapped physical device, and then
+// jump to the next function in the call chain
+
+.include "gen_defines.asm"
+
+.if AARCH_64
+
+.macro PhysDevExtTramp num
+.global vkPhysDevExtTramp\num
+vkPhysDevExtTramp\num:
+    ldr     x9, [x0]                                                 // Load the loader_instance_dispatch_table* into x9
```

Comment:
x9/x10/x11 are overwritten, but they are not recovered, so it will cause problems either, x9/x10/11 could be used by the compiler before call this method, i.e. if there are more than 10 parameters, compiler will store parameters into x9/x10/11.
So can we choose x20 or higher registers and store/recovered as well?

Reply:
It should be okay to use those registers without saving and restoring their state. The aarch64 ABI specifies registers x0-x7 for use as argument registers. If any more arguments are required they need to be placed on the stack. Registers x9-x15 are temporary registers that a subroutine can overwrite at will and are the responsibility of the caller to save and restore if they need to be preserved after a subroutine.

https://developer.arm.com/documentation/den0024/a/The-ABI-for-ARM-64-bit-Architecture/Register-use-in-the-AArch64-Procedure-Call-Standard/Parameters-in-general-purpose-registers

Reply:
Make sense and it looks good to me.